src/server/game/Movement/MovementGenerators/PointMovementGenerator.cpp

```diff
@@ -181,6 +181,24 @@ template void PointMovementGenerator<Creature>::DoFinalize(Creature*, bool, bool
 
 //---- AssistanceMovementGenerator
 
+void AssistanceMovementGenerator::Initialize(Unit* owner)
+{
+    RemoveFlag(MOVEMENTGENERATOR_FLAG_INITIALIZATION_PENDING | MOVEMENTGENERATOR_FLAG_DEACTIVATED);
+    AddFlag(MOVEMENTGENERATOR_FLAG_INITIALIZED);
+
+    if (owner->HasUnitState(UNIT_STATE_LOST_CONTROL | UNIT_STATE_NOT_MOVE))
+        return;
+
+    if (!owner->IsStopped())
+        owner->StopMoving();
+
+    owner->AddUnitState(UNIT_STATE_ROAMING | UNIT_STATE_ROAMING_MOVE);
+    Movement::MoveSplineInit init(owner);
+    init.MoveTo(_x, _y, _z);
+    init.SetVelocity(owner->GetSpeed(MOVE_RUN) * 0.66);
```

Comment:
This smells like a ass ugly hack to me. Where did you pull this number from?

Reply:
The number is from the issue comments and it seems reasonable, feels like what's in classic wow.

Reply:
Already done here with the same value: https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Entities/Unit/Unit.cpp#L8490-L8491

Reply:
Does it make much difference with or without `speed *= 0.66f` (float indicator)?